### PR TITLE
Multiple and one book to multiple stores

### PR DESCRIPTION
**Pull request by BenjyJack**

### CODE REVIEW

File: datastore/src/main/java/com/ds/datastore/BookStoreController.java [hunk 1]
```diff
@@ -73,6 +71,10 @@ private void registerWithHub() throws Exception {
         JsonObject json = new JsonObject();
         json.addProperty("id", this.id);
         json.addProperty("address", this.url + "/bookstores/" + this.id);
+        outputJson(con, gson, json);
+    }
+
+    protected static void outputJson(HttpURLConnection con, Gson gson, JsonObject json) throws IOException {
```
Comment (yspotts) [inline datastore/src/main/java/com/ds/datastore/BookStoreController.java]: I think this would be best in a separate utility class.

File: datastore/src/main/java/com/ds/datastore/BookController.java [hunk 2]
```diff
@@ -50,6 +58,78 @@ public BookController(BookRepository repository, BookModelAssembler assembler, B
...
+                url = new URL(map.get(Long.parseLong(storeId)) + "/books");
+            } catch(Exception e){
+                continue;
+            }
+            HttpURLConnection con = createConnection(this.map.get(Long.parseLong(storeId)), "POST");
+
+            Gson gson = new Gson();
+            JsonObject jso = makeJson(book);
+            BookStoreController.outputJson(con, gson, jso);
+
```
Comment (yspotts) [inline datastore/src/main/java/com/ds/datastore/BookController.java]: these extra lines are driving me crazy :)

File: datastore/src/main/java/com/ds/datastore/BookController.java [hunk 3]
```diff
@@ -50,6 +58,78 @@ public BookController(BookRepository repository, BookModelAssembler assembler, B
...
+
+            entityList.add(assembler.toModel(new Book(book)));
+        }
+
+
+        return CollectionModel.of(entityList, linkTo(methodOn(BookController.class).oneBookToManyStores(null, null)).withSelfRel());
+    }
+
+    private JsonObject makeJson(Book book) {
```
Comment (yspotts) [inline datastore/src/main/java/com/ds/datastore/BookController.java]: I think it makes more sense to put this on the `Book` class.

File: datastore/src/main/java/com/ds/datastore/Book.java [hunk 4]
```diff
@@ -28,6 +28,15 @@
     private Long storeID;
 
     public Book() {}
+    public Book(Book book)
```
Comment (yspotts) [inline datastore/src/main/java/com/ds/datastore/Book.java:33]: why is this necessary?

File: datastore/src/main/resources/application.properties [hunk 5]
```diff
@@ -6,5 +6,5 @@ spring.datasource.password=
 #spring.jpa.show-sql=true
 spring.jpa.hibernate.ddl-auto=update
 application.baseUrl=${ADDRESS}
-#server.port= ${PORT}
+server.port= ${PORT}
```
Comment (yspotts) [inline datastore/src/main/resources/application.properties]: I think its best if we remove the space before the `$` on this line and the next

File: datastore/src/main/java/com/ds/datastore/BookController.java [hunk 6]
```diff
@@ -65,23 +61,17 @@ public BookController(BookRepository repository, BookModelAssembler assembler, B
         for(String storeId: id)
         {
             book.setStoreID(Long.parseLong(storeId));
-            URL url;
             try{
-                url = new URL(map.get(Long.parseLong(storeId)) + "/books");
+                URL url = new URL(map.get(Long.parseLong(storeId)) + "/books");
```
Comment (yspotts) [inline datastore/src/main/java/com/ds/datastore/BookController.java]: Im not clear about the point of this? If you want to check if the map has the storeId, why not just check that directly?

File: datastore/src/main/java/com/ds/datastore/Book.java (hunk 4 above)
Comment (TGfor3) [reply datastore/src/main/java/com/ds/datastore/Book.java:33]: Since java is pass by reference for the one to many stores method we wanted to add the same book with different store ids to a list, in order to make copies without getting and setting each variable, we now have a constructor which creates it

Comment (TGfor3) [reply datastore/src/main/java/com/ds/datastore/Book.java:33]: ^From Seth

Comment (yspotts) [reply datastore/src/main/java/com/ds/datastore/Book.java:33]: 👍

**Review by yspotts — APPROVED** [review]
